test/NearCacheFailoverTest.cpp

```diff
@@ -0,0 +1,191 @@
+#include "infinispan/hotrod/ConfigurationBuilder.h"
+#include "infinispan/hotrod/RemoteCacheManager.h"
+#include "infinispan/hotrod/RemoteCache.h"
+#include "infinispan/hotrod/Version.h"
+
+#include "infinispan/hotrod/JBasicMarshaller.h"
+#include "infinispan/hotrod/CacheClientListener.h"
+#include "infinispan/hotrod/ClientEvent.h"
+
+#include <chrono>
+#include <thread>
+#include <string>
+#include <cstdlib>
+using namespace infinispan::hotrod;
+using namespace infinispan::hotrod::event;
+
+// This balancing strategy always returns server 11322 if up
+// then fallback on 11222
+class StickyBalancingStrategy: public FailOverRequestBalancingStrategy {
+public:
+    StickyBalancingStrategy() :
+            index(0) {
+    }
+    static FailOverRequestBalancingStrategy *newInstance() {
+        return new StickyBalancingStrategy();
+    }
+
+    void setServers(const std::vector<transport::InetSocketAddress> &s) {
+        servers = s;
+        // keep the old index if possible so that we don't produce more requests for the first server
+        if (index >= servers.size()) {
+            index = 0;
+        }
+    }
+
+    ~StickyBalancingStrategy() {
+    }
+
+    const transport::InetSocketAddress &getServerByIndex(size_t pos) {
+        const transport::InetSocketAddress &server = servers[pos];
+        return server;
+    }
+private:
+    std::vector<transport::InetSocketAddress> servers;
+    size_t index;
+    const transport::InetSocketAddress &nextServer(
+            const std::set<transport::InetSocketAddress>& failedServers) {
+        if (servers.size() == 1)
+            return servers[0];
+        if (failedServers.size() == 0)
+                {
```

Comment:
Just some formatting issues in this file. Sometimes brackets are on the same line, and sometimes not. Plus some spacing issues as well.